### ISSUE TITLE
Ignore option on the require-jsdoc rule

### ISSUE THREAD

**Issue by AvraamMavridis** [eslint/eslint#5567]
( I am using "eslint": "^1.9.0" )
It will be nice if we can specify functions that the rule will ignore. e.g. the `render` function on React components is kinda obvious what is doing and a comment block is not really needed.

![screen shot 2016-03-14 at 10 58 36](https://cloud.githubusercontent.com/assets/14922483/13741037/34b779ae-e9d4-11e5-82b5-a4d41e756c59.png)

It will be nice if we can have something like:

```js
        "require-jsdoc"              : [ 2,
        {
            "require":
            {
                "FunctionDeclaration" : true,
                "MethodDefinition" : true,
                "ClassDeclaration" : true
            },
            "ignore":
            {
                "render"  : true,
                "constructor" : true,
                "componentDidMount"  : true
            }
        }],
```

**Comment by eslintbot** [comment]
Hi @AvraamMavridis, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
@AvraamMavridis In case the comment above wasn't clear, we need a little more information about exactly how your proposed rule augmentation would work. I see one example in your initial post, but we like to see examples (with code) of what would constitute a rule violation and what wouldn't under your new proposal.

The [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) guidelines might be useful too (except in this case, you're proposing a rule enhancement, not a new rule). Take a look at those and use that as a guide for providing us the information we need.

Thanks for submitting your issue, and I look forward to seeing what you come up with.

**Comment by AvraamMavridis** [comment]
@platinumazure I see. I will provide some more examples.

**Comment by nzakas** [comment]
This would be a pretty big hack and would likely have a lot of false positives, so I don't think we can implement this.

**Comment by UserBug** [comment]
@nzakas Thank you for your answer.
I think there's a lot of libraries (not only React) which contain good known interfaces.
Always add JSdoc to implementation of those functions and methods may be expensive. 
I don't see any logic to write:
/** render */
/** render */
/** render */
... Before typical method.
But, of course, I understand that JSdoc must be required for custom functions and methods.
So add "ignore" config with array of function names will be very helpful.

Example: 
`
"require-jsdoc"              : [ 2,
        {
            "require":
            {
                "FunctionDeclaration" : true,
                "MethodDefinition" : true,
                "ClassDeclaration" : true
            },
            "ignore":
            {
                "MethodDefinition" : ["render", "constructor", "componentDidMount"],
            }
        }],
`

**Comment by websirnik** [comment]
I also would benefit from this option. You can similar function declaration patterns in Angular as well.  It is very common to have `function activate()` or `function Controller()` in every component. Those should not require jsDoc.

**Comment by JakubRimal** [comment]
There is a quite new plugin for that: https://github.com/MaienM/eslint-plugin-require-jsdoc-except

It even supports regular expressions so you can add e.g. ignore rule `"/^_.*$/"` which will ignore require-jsdoc for all method/functions starting with `_`.
